core-bundle/src/Migration/Version408/Version480Update.php

```diff
@@ -144,7 +209,45 @@ public function run(): MigrationResult
             SET
                 skipIfDimensionsMatch = '1'
         ");
+    }
+
+    public function shouldRunImportantPart(): bool
+    {
+        $schemaManager = $this->connection->getSchemaManager();
+
+        if (!$schemaManager->tablesExist(['tl_files'])) {
+            return false;
+        }
 
+        $columns = $schemaManager->listTableColumns('tl_files');
+
+        if (
+            !isset(
+                $columns['path'],
+                $columns['importantpartx'],
+                $columns['importantparty'],
+                $columns['importantpartwidth'],
+                $columns['importantpartheight']
+            )
+        ) {
+            return false;
+        }
+
+        return (bool) $this->connection->query('
+            SELECT EXISTS(
+                SELECT id
+                FROM tl_files
+                WHERE
+                    importantPartX > 1
+                    OR importantPartY > 1
+                    OR importantPartWidth > 1
+                    OR importantPartHeight > 1
+            )
+        ')->fetchColumn();
```

Comment:
In theory there could be edge cases where the important part is still absolute instead of relative - yet has a coordinate of `0,0` and a width of `0,0` or `1,0` or `1,1` 😬

Reply:
I know :(
Do you think we should instead check if the column type is an integer (instead of `DOUBLE`). But this would not work then for everyone that has already converted the data type but still has the old “absolute” values.

Reply:
Or maybe better: If the column type is integer we migrate, otherwise we only migrate if there are actual values that are integer 2 or higher...

Reply:
I think that would be the way to go yes. However there might still be edge cases where the migration failed (for whatever reason) _after_ the database field type was already altered. At this point you still have the old absolute integer values in the new double database field.

But I think that is the best we can do for automatic detection.

Reply:
This should be better now with 325525c5144939ec568f85a58be50bea878b2399

Reply:
I think/hope I got all edge cases now in d74fcec5bc5fb430d05a1bfbd96a08c404cba912